### PR TITLE
Handle Roles not being displayed properly in users list

### PR DESCRIPTION
**Pull request by Sameh16**
# Changes proposed ✍️

[Screencast from 2024-12-11 17-39-31.webm](https://github.com/user-attachments/assets/8fcd4183-291b-4e90-96ed-61ae7d8b1765)

### What
Handle Roles not being displayed properly in users list

## Checklist ✅
- [x] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [x] Add screenshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `UserRole` enumeration with roles: `admin`, `projectAdmin`, and `readonly`.
	- Enhanced user interface for displaying user roles with tooltips and popovers for additional context.

- **Improvements**
	- Updated user data handling to include `adminSegments` for active users.
	- Refined role display logic to accommodate multiple user roles.

- **Bug Fixes**
	- Minor adjustments made to the empty state component and loading spinner for consistency.

### CODE REVIEW

File: frontend/src/modules/admin/modules/users/pages/users.page.vue [hunk 1]
```diff
@@ -117,17 +162,32 @@ const loadMore = () => {
   fetchUsers();
 };
 
-const roleDisplay = (roles: string[]) => {
-  const role = roles?.[0];
+const projects = computed(() => treeMap(segmentStore.projectGroups.list));
```
Comment (gaspergrom) [inline frontend/src/modules/admin/modules/users/pages/users.page.vue:165]: We have a function in segment.js to get segment name by id so you dont need to do this stuff.

File: frontend/src/modules/admin/modules/users/pages/users.page.vue [hunk 2]
```diff
@@ -28,30 +34,63 @@
...
+                    <div class="p-1">
+                      <span class="text-xs text-gray-400">Project groups</span>
+                      <div class="overflow-auto max-h-30 mt-4 flex items-baseline flex-wrap gap-2">
+                        <p
+                          v-for="segmentId in user.adminSegments"
+                          :key="segmentId"
+                          class="badge--border !block badge--gray-light h-6 text-xs"
+                        >
+                          {{
+                            projects.find(
```
Comment (gaspergrom) [inline frontend/src/modules/admin/modules/users/pages/users.page.vue:72]: Here use function getSegmentName from segment.js

File: frontend/src/modules/admin/modules/users/pages/users.page.vue [hunk 3]
```diff
@@ -28,30 +34,63 @@
...
+                        </p>
+                      </div>
+                    </template>
+                    <div class="p-1">
+                      <span class="text-xs text-gray-400">Project groups</span>
+                      <div class="overflow-auto max-h-30 mt-4 flex items-baseline flex-wrap gap-2">
+                        <p
+                          v-for="segmentId in user.adminSegments"
+                          :key="segmentId"
+                          class="badge--border !block badge--gray-light h-6 text-xs"
```
Comment (gaspergrom) [inline frontend/src/modules/admin/modules/users/pages/users.page.vue:69]: Use badge component lf-badge here

**Review by gaspergrom — CHANGES_REQUESTED** [review]

File: frontend/src/modules/admin/modules/users/pages/users.page.vue (hunk 2 above)
Comment (Sameh16) [reply frontend/src/modules/admin/modules/users/pages/users.page.vue]: Done

File: frontend/src/modules/admin/modules/users/pages/users.page.vue (hunk 3 above)
Comment (Sameh16) [reply frontend/src/modules/admin/modules/users/pages/users.page.vue]: Done

**Review by gaspergrom — APPROVED** [review]

**Pull request merged**